types/cavy/package.json

```diff
@@ -10,7 +10,7 @@
     },
     "devDependencies": {
         "@types/cavy": "workspace:.",
-        "@types/react-native": "*"
+        "@types/react-native": "^0.72"
```

Comment:
I believe that these should all now depend on `react-native` directly, or, leave this as-is as it'll automatically pull in the shim?

Reply:
Hmm I guess for testing we should probably point it to `react-native`, but I'm concerned about potential breaking changes for new types if we continue with `*` -- `react-native` doesn't truly follow semver so every minor may contain breaking changes. 

In the past when I cut a new minor version of `types/react-native`, if there was a package that broke with the types, I'd fix it at the last version, that's my inclination here. By that logic, this should be `"~0.72"`, I can update to that instead. What do you think @jakebailey?

Reply:
Pinning will mean that anyone who installs any of these packages will get multiple copies of `react-native` types in their install without a way to fix that, besides using package manager overrides to force overwrite these. If these packages don't specify a range of `react-native` themselves, then I would think we wouldn't try and pin them here unless the break was unavoidable. If you've been pinning them back, then maybe that's okay and precedented, but it just seems like a bad time for downstream users.

`^0.72` is equivalent to `~0.72` as the major version is a zero, so there's no difference there.

Reply:
@andrewbranch @sandersn do you have any notion of whether or not old packages should depend on `@types/react-native` or `react-native` itself right away? The former may be better just because the deprecation shim will make sure the real thing is installed... We unfortunately can't really test with the shim locally, though.

Reply:
react-router has been shipping its own types for a while now, which cause conflicts with `@types/react`, so it seems like it would be better to get `@types/react` out of the picture as often as possible.